src/atl/formula/parser.rs

```diff
@@ -0,0 +1,827 @@
+use std::str::{self, FromStr};
+use std::sync::Arc;
+
+use pom::parser::{end, Parser};
+use pom::parser::{list, one_of, seq, sym};
+
+use super::Phi;
+
+/// Parse an ATL formula
+pub(crate) fn parse_phi<'a, 'b: 'a, A: ATLExpressionParser>(
+    expr_parser: &'b A,
+    input: &'a str,
+) -> Result<Phi, String> {
+    let formula = ws() * phi(expr_parser) - end();
+    formula
+        .parse(input.as_bytes())
+        .map_err(|err| err.to_string())
+}
+
+/// Allows a CGS model to define custom player and proposition expressions. For instance,
+/// in LCGS we want to be able to write "p2" as a player and "p2.alive" as a proposition, while
+/// in json, players and propositions are numbers.
+pub trait ATLExpressionParser {
+    /// A parser that parses a player name
+    fn player_parser(&self) -> Parser<u8, usize>;
+    /// A parser that parses a proposition name
+    fn proposition_parser(&self) -> Parser<u8, usize>;
```

Comment:
This should return `Parser<u8, crate::common::Proposition>`. `Proposition` is a alias for `usize` but having a use-case specific name for it helps if we ever need to refactor.

Reply:
Good idea. Fixed in 0d67118

Reply:
Now that I think about, `player_parser` should probably return `crate::common::Player`